java/test/src/main/java/io/ray/test/FailureTest.java

```diff
@@ -98,6 +128,27 @@ public void testActorCreationFailure() {
     assertTaskFailedWithRayTaskException(actor.task(BadActor::badMethod).remote());
   }
 
+  public void testActorCreationFailedAndRestarted() {
+    String flagFileName = "/tmp/TEST_ERROR_FLAG_" + new Random().nextInt(9999);
+    ActorHandle<ActorFailedAtFirstTime> actor =
+        Ray.actor(ActorFailedAtFirstTime::new, flagFileName).setMaxRestarts(2).remote();
+    // At first, actor will throw exceptions and restarted.
+    // Finally after restarted, actor will be created successfully.
+    boolean ok =
+        TestUtils.waitForCondition(
+            () -> {
+              try {
+                Assert.assertEquals(actor.task(ActorFailedAtFirstTime::ok).remote().get(), "OK");
```

Comment:
We can return `Ray.getRuntimeContext().wasCurrentActorRestarted()` in the "ok" method and assert the result.

Reply:
The "ok" method is for checking whether the actor is alive for method calls. Return a constant string is enough. I think It's not necessary to assert wasCurrentActorRestarted() in this method.